jfr-connection/src/main/java/io/opentelemetry/contrib/jfr/connection/Recording.java

```diff
@@ -270,8 +270,14 @@ public void close() throws IOException, JfrConnectionException {
         connection.stopRecording(id);
       } catch (IOException | JfrConnectionException ignored) {
         // Stopping the recording is best-effort
-      } finally {
+      }
+    }
+    if (oldState == State.STOPPED || oldState == State.RECORDING) {
+      try {
         connection.closeRecording(id);
+      } catch (IOException | JfrConnectionException | UnsupportedOperationException ignored) {
+        // Closing the recording is best-effort
+        // FlightRecorderDiagnosticCommandConnection close throws UnsupportedOperationException
```

Comment:
Consider logging a warning. This shouldn't happen very often, and if it does it's nice to have some indication or remnant of a deeper underlying problem. Root cause analysis can be difficult without this.